test/NuGet.Core.FuncTests/Dotnet.Integration.Test/MsbuildIntegrationTestFixture.cs

```diff
@@ -248,123 +261,106 @@ private void CopyLatestCliToTestDirectory(string destinationDir)
+                        sourceFileName: Path.Combine(packProjectCoreArtifactsDirectory.FullName, "ilmerge", packFileName),
+                        destFileName: Path.Combine(packAssemblyDestinationDirectory, packFileName));
                 }
-                catch
+                else
                 {
-                    // Do nothing
-                }
-                using (var stream = entry.Open())
-                {
-                    var copiedFile = packageFileExtractor.ExtractPackageFile(packageFileName, targetFilePath, stream);
-                    if (copiedFile != null)
+                    foreach (var assembly in packProjectCoreArtifactsDirectory.EnumerateFiles("*.dll"))
                     {
-                        entry.UpdateFileTimeFromEntry(copiedFile, logger);
-
-                        filesCopied.Add(copiedFile);
+                        File.Copy(
+                            sourceFileName: assembly.FullName,
+                            destFileName: Path.Combine(packAssemblyDestinationDirectory, assembly.Name),
+                            overwrite: true);
                     }
                 }
+                // Copy the pack targets
+                var packTargetsSource = Path.Combine(packProjectCoreArtifactsDirectory.FullName, packTargetsName);
+                var targetsDestination = Path.Combine(pathToPackSdk, "build", packTargetsName);
+                var targetsDestinationCrossTargeting = Path.Combine(pathToPackSdk, "buildCrossTargeting", packTargetsName);
+                File.Copy(packTargetsSource, targetsDestination, overwrite: true);
+                File.Copy(packTargetsSource, targetsDestinationCrossTargeting, overwrite: true);
             }
-
-        }
-
-        private void CopyNupkgFilesToTarget(PackageArchiveReader nupkg, string destPath, IEnumerable<string> files)
-        {
-            var packageFileExtractor = new PackageFileExtractor(files,
-                                         PackageExtractionBehavior.XmlDocFileSaveMode);
-
-            nupkg.CopyFiles(destPath, files, packageFileExtractor.ExtractPackageFile, new TestCommandOutputLogger(),
-                CancellationToken.None);
-
-        }
-
-        private static string FindMostRecentNupkg(string nupkgDirectory, string id)
-        {
-            var info = LocalFolderUtility.GetPackagesV2(nupkgDirectory, new TestLogger());
-
-            return info.Where(t => t.Identity.Id.Equals(id, StringComparison.OrdinalIgnoreCase))
-                .Where(t => !Path.GetExtension(Path.GetFileNameWithoutExtension(t.Path)).Equals(".symbols"))
-                .OrderByDescending(p => p.LastWriteTimeUtc)
-                .First().Path;
         }
 
         public void Dispose()
         {
-            RunDotnet(Path.GetDirectoryName(TestDotnetCli), "build-server shutdown");
-            KillDotnetExe(TestDotnetCli);
-            _cliDirectory.Dispose();
+            if (RuntimeEnvironmentHelper.IsWindows)
```

Comment:
I don't understand why this check exists.

The `TempDirectory`'s below are created on all platforms and should be disposed unconditionally.

Reply:
Fixed. I had that deleted, but somehow I messed something up in one of my rebases.